### ISSUE TITLE
No confmap support in releases

### ISSUE THREAD

**Issue by gillg** [open-telemetry/opentelemetry-collector-contrib#29345]
### Component(s)

confmap/provider/s3provider

## Description
Correct me if I'm wrong, but I have the feeling since these manifest has been introduced https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol/manifest.yaml & https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/manifest.yaml the confmap https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/confmap/provider/s3provider seems not usable at all. Probably others too.
I always have an error "unsupported scheme", I also tried the feature flag which is deprectated.

## Steps to Reproduce
Just try `--config s3://xxxxxxx/xxxx.yml`

## Actual Result
```
Feature gate "confmap.expandEnabled" is stable and already enabled. It will be removed in version 0.75.0 and continued use of the gate after version 0.75.0 will result in an error.
Error: unsupported scheme on URI "s3://xxxxxxxxxx.s3.eu-west-3.amazonaws.com/otel-config/generic.yml"
2023/11/20 07:53:42 collector server run finished with error: unsupported scheme on URI "s3://xxxxxxxxxx.s3.eu-west-3.amazonaws.com/otel-config/generic.yml"
```

### Collector version

0.89.0

## Environment
OS: al2023 ARM64

### OpenTelemetry Collector configuration

```yaml
N/A
```

### Log output

```shell
N/A
```

### Additional context

N/A

**Comment by crobert-1** [comment]
**Debugging**

Reproduces everytime, here's the call stack I'm getting from hitting in a debugger:
```
confmap.NewResolver (resolver.go:95) go.opentelemetry.io/collector/confmap
otelcol.NewConfigProvider (configprovider.go:86) go.opentelemetry.io/collector/otelcol
otelcol.newCollectorWithFlags (command.go:44) go.opentelemetry.io/collector/otelcol
otelcol.NewCommand.func1 (command.go:23) go.opentelemetry.io/collector/otelcol
cobra.(*Command).execute (command.go:983) github.com/spf13/cobra
cobra.(*Command).ExecuteC (command.go:1115) github.com/spf13/cobra
cobra.(*Command).Execute (command.go:1039) github.com/spf13/cobra
main.runInteractive (main.go:27) main
main.run (main_others.go:11) main
main.main (main.go:20) main
runtime.main (proc.go:267) runtime
runtime.goexit (asm_amd64.s:1650) runtime
```
It's actually independent of the format of the URI. This is being hit in collector startup.

From the call stack line: 
```
otelcol.newCollectorWithFlags (command.go:44) go.opentelemetry.io/collector/otelcol
``` 
There's a method call [here](https://github.com/open-telemetry/opentelemetry-collector/blob/0d1d24475a18bfa1020802c36df1002535b70432/otelcol/command.go#L44
) from earlier in the function:
```
		set.ConfigProvider, err = NewConfigProvider(newDefaultConfigProviderSettings(configFlags))
```
Which calls into [here](https://github.com/open-telemetry/opentelemetry-collector/blob/0d1d24475a18bfa1020802c36df1002535b70432/otelcol/configprovider.go#L134):
```
func newDefaultConfigProviderSettings(uris []string) ConfigProviderSettings {
	return ConfigProviderSettings{
		ResolverSettings: confmap.ResolverSettings{
			URIs:       uris,
			Providers:  makeMapProvidersMap(fileprovider.New(), envprovider.New(), yamlprovider.New(), httpprovider.New(), httpsprovider.New()),
			Converters: []confmap.Converter{expandconverter.New()},
		},
	}
}
```
When we're checking for the scheme `s3` it checks against this list of providers. Since the s3 provider is not listed here, the collector is failing to find a provider.

**Conclusion**

Running the contrib collector distribution and simply specifying a config path starting with `s3` fails to find the s3 config provider. Is there more configuration required here to get the s3 provider added to the list map of valid schemes and scheme providers? Or is this a valid bug that needs addressed?

**Comment by crobert-1** [comment]
It looks like it's simply not supported in `contrib` or `core`. Other distributions support it by manually adding it to the list of valid providers. I found an AWS [support document](https://aws-otel.github.io/docs/components/confmap-providers#s3-provider) referencing this, and found that the [AWS distribution does this.](https://github.com/aws-observability/aws-otel-collector/blob/6822a4346b136e71bfabdac7ff9fe36c4a743e3b/pkg/config/config_factory.go#L47)

@Aneurysm9: Should we be adding support for the s3 provider to core/contrib? This may require moving the s3 provider to core, and enabling it by default.

Feel free to correct me if I'm wrong here, I may be misunderstanding something simple here.

**Comment by bryan-aguilar** [comment]
From what I can tell so far your analysis looks to be correct @crobert-1. 

> Should we be adding support for the s3 provider to core/contrib? This may require moving the s3 provider to core, and enabling it by default.

I think this can be a more general question. Should the contrib distibution support different config map providers? 

If the answer is yes then I think this problem should be solved generically by expanding the [ocb](https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/builder/README.md) tool.

**Comment by crobert-1** [comment]
Good point. It seems like config map providers may be a bit of a gray area in components. This is currently the only config map provider in contrib, but it has a label as well as code owners. However, there's no `metadata.yaml` file to specify stability level, and there's no comment about it being included in any distribution of the collector.

I guess it's up to code owners to know if this could be considered `Alpha` [stability](https://github.com/open-telemetry/opentelemetry-collector#alpha). If it is, it would be good to make this a more full-fledged component in contrib (with a `metadata.yaml` file) and then add it to the collector builder.

In my mind, if the component is entirely in contrib and more stable than `Development`, I don't see any reason why it shouldn't be included in a release. Anyone's welcome to correct me here, I may be missing something.

**Comment by bryan-aguilar** [comment]
IMO, this should not be in development stability anymore and is most likely an oversight after original contribution. We (ADOT) already include this in our distribution and provide support for this. 

> If it is, it would be good to make this a more full-fledged component in contrib (with a metadata.yaml file) and then add it to the collector builder.

We should probably do this first, as I'm not sure if mdatagen supports confmap components yet. This may need to be captured in a separate issue first. I'll take a look at getting a `metadata.yaml` file in there.

**Comment by bryan-aguilar** [comment]
Looks like I missed it on first pass but it already has a [metadata.yaml](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/confmap/provider/s3provider/metadata.yaml) file but stability level is not defined yet.

**Comment by crobert-1** [comment]
> Looks like I missed it on first pass but it already has a [metadata.yaml](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/confmap/provider/s3provider/metadata.yaml) file but stability level is not defined yet.

I totally missed it myself as well. I think it's because there's no header in the README, but that'll be solved by the issue you've filed 👍

**Comment by crobert-1** [comment]
_I'm going to remove `needs triage` as the plan is to add support for this in the `contrib` repo._

**Comment by bryan-aguilar** [comment]
Agreed, I think we will also need to make changes to the opentelemetry collector builder to support this but I have not evaluated changes needed there.

**Comment by gillg** [comment]
Someone to reopen?
Cc @atoulme

**Comment by atoulme** [comment]
I believe the collector supports providers now. I didn't try this yet.

**Comment by Keimille** [comment]
> I believe the collector supports providers now. I didn't try this yet.

I am still having this issue on the latest version, unless I am missing something altogether.

**Comment by evan-bradley** [comment]
There's still some requisite work to be done on mdatagen to promote the providers to alpha stability, but the S3 provider will be added to the contrib distro in this PR: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/546.

In the meantime, you can still add the provider to your OCB manifests if you are using a custom distro [like this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2997a9018fb4bdfd21f35c890c1fa6c0cc532365/cmd/otelcontribcol/builder-config.yaml#L240).

**Comment by garysassano** [comment]
I encountered the same issue. For now, I'm using the HTTPS confmap provider with the S3 object path as a workaround until the S3 confmap provider is officially added. Alternatively, you could use the ADOT distribution, which already supports the S3 confmap provider.

**Comment by ksandrmatveyev** [comment]
Hello colleagues. @evan-bradley, Could you clarify if that feature was implemented (according to https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/confmap/provider/s3provider and https://github.com/open-telemetry/opentelemetry-collector-releases/pull/546) and if the status is wrong for the current issue (`Not planned`)?

**Comment by evan-bradley** [comment]
Closing as resolved via https://github.com/open-telemetry/opentelemetry-collector-releases/pull/546.
